migrations/utils.py

```diff
@@ -0,0 +1,208 @@
+"""Double Linked-List
+
+Defines a specialized doubly linked list implementation restricted to unique data 
+values and no cycles to prevent migration conflicts. Holds no cycles and raises
+errors when duplicate values are attempted to be added. Is intended to be re-build
+each time new migration is added to the migrations directory. 
+
+Self.head is always pointing to the latest created migration id, tail is pointing
+to the first created migration (the migration with no preceding migration).
+Hence, primary iterates backwards and adds forward.
+"""
+class Node:
+    def __init__(self, data):
+        """Initialize a node with two connections containing data."""
+        self.data = data
+        self.prev_node = None
+        self.next_node = None
+
+    def __repr__(self):
+        """Representation of the node."""
+        return f"{self.data}"
+
+    def __eq__(self, other):
+        """Check equality of nodes."""
+        if isinstance(other, Node):
+            return self.data == other.data
+        return False
+
+    def __hash__(self):
+        """Hash the node's data."""
+        return hash(self.data)
+
+
+class LinkedList:
+    def __init__(self):
+        """Initialize a linked list."""        
+        self.head = None
+
+    def find(self, data) -> Node:
+        """Find the first node containing specified data."""
+        current_node = self.head
+        while current_node:
+            if current_node.data == data:
+                return current_node
+            current_node = current_node.prev_node
+        return None
+
+    def present(self, data) -> bool:
```

Comment:
I think you could override `__contains__()` so you could use the `in` keyword to make it slightly more idiomatic.
I don't see this invoked anywhere, so it's only a nitpick